scene/gui/scroll_container.cpp

```diff
@@ -282,7 +282,7 @@ void ScrollContainer::ensure_control_visible(Control *p_control) {
 	float right_margin = v_scroll->is_visible() ? v_scroll->get_size().x : 0.0f;
 	float bottom_margin = h_scroll->is_visible() ? h_scroll->get_size().y : 0.0f;
 
-	Vector2 diff = Vector2(MAX(MIN(other_rect.position.x, global_rect.position.x), other_rect.position.x + other_rect.size.x - global_rect.size.x + (!is_layout_rtl() ? right_margin : 0.0f)),
+	Vector2 diff = Vector2(MAX(MIN(other_rect.position.x - (is_layout_rtl() ? right_margin : 0.0f), global_rect.position.x), other_rect.position.x + other_rect.size.x - global_rect.size.x + (!is_layout_rtl() ? right_margin : 0.0f)),
```

Comment:
This is pretty minor, but I think it might be more clear to remain `right_margin` to `side_margin`. Since it's used to adjust the position based on the presence of a side scrollbar, perhaps `side_margin` would be more descriptive.

Reply:
That makes sense.